### PR TITLE
Docs/incremental import

### PR DESCRIPTION
**Pull request by MrCheater**

### CODE REVIEW

File: docs/advanced-techniques.md [hunk 1]
```diff
@@ -113,3 +113,92 @@ The **resolve-module-uploader** module adds the following API endpoints to an ap
...
+const eventStore2 = createEventStoreAdapter({
+  databaseFile: './data/event-store-2.db'
+})
+
+await pipeline(eventStore1.export(), eventStore2.import())
+```
+
+## Incremental import
+
+The incremental import allows you to import into an event store only those events that do not already exist in this event store. The incremental import also skips events that are older (have an older timestamp) than the latest event in the recipient event store.
```
Comment (albertov05) [inline docs/advanced-techniques.md:150]:
```suggestion
Incremental import allows you to import into an event store only those events that do not already exist in this event store. Incremental import also skips events that are older (i.e., have an older timestamp) than the latest event in the recipient event store.
```

File: docs/advanced-techniques.md [hunk 2]
```diff
@@ -113,3 +113,92 @@ The **resolve-module-uploader** module adds the following API endpoints to an ap
...
+
+### Basic Incremental Import
+
+To import events incrementally, pass an array of events to an event store adapter's [incrementalImport](api-reference.md#incrementalimport) method.
+
+The code sample below demonstrates how to implement and API endpoint that incrementally imports events into the application's event store.
```
Comment (albertov05) [inline docs/advanced-techniques.md:156]:
```suggestion
The code sample below implements an API endpoint that incrementally imports events into the application's event store.
```

File: docs/api-reference.md [hunk 3]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
 
 <!-- prettier-ignore-end -->
 
+## Event Store Adapter API
+
+An event store adapter defines how the reSolve framework stores events in the underlying event store. An event store adapter object must expose the following functions:
```
Comment (albertov05) [inline docs/api-reference.md:410]: Is it "reSolve" or "ReSolve"? Please make sure that capitalization is consistent throughout the document.

File: docs/api-reference.md [hunk 4]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+
+| Function Name                                           | Description                                                                                           |
+| ------------------------------------------------------- | ----------------------------------------------------------------------------------------------------- |
+| [init](#init)                                           | Initializes a database.                                                                               |
+| [drop](#drop)                                           | Drops a database.                                                                                     |
+| [dispose](#dispose)                                     | Disconnects from a database and disposes unmanaged resources.                                         |
+| [saveEvent](#saveevent)                                 | Saves an event to the database.                                                                       |
+| [loadEvents](#loadEvents)                               | Gets an array of events and the next cursor from the store based on the specified filtering criteria. |
```
Comment (albertov05) [inline docs/api-reference.md:418]:
```suggestion
| [loadEvents](#loadEvents)                               | Gets an array of events and the next cursor from the store based on the specified filter criteria. |
```

File: docs/api-reference.md [hunk 5]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+| [getLatestEvent](#getlatestevent)                       | Gets the latest saved event.                                                                          |
+| [import](#import)                                       | Gets a writable stream used to save events.                                                           |
+| [export](#export)                                       | Gets a readable stream used to load events.                                                           |
+| [freeze](#freeze)                                       | Freezes the database.                                                                                 |
+| [unfreeze](#unfreeze)                                   | Unfreezes the database.                                                                               |
+| [isFrozen](#isfrozen)                                   | Gets a boolean value that indicates whether the database is frozen.                                   |
```
Comment (albertov05) [inline docs/api-reference.md:424]: "Boolean" should always be capitalized:

```suggestion
| [isFrozen](#isfrozen)                                   | Gets a Boolean value that indicates whether the database is frozen.                                   |
```

File: docs/api-reference.md [hunk 6]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+| [deleteSecret](#deletesecret)                           | Deletes a secret.                                                                                     |
+| [incrementalImport](#incrementalimport)                 | Incrementally imports events.                                                                         |
+| [beginIncrementalImport](#beginincrementalimport)       | Starts to build a batch of events to import.                                                          |
+| [pushIncrementalImport](#pushincrementalimport)         | Adds events to an incremental import batch.                                                           |
+| [commitIncrementalImport](#commitincrementalimport)     | Commits an incremental import batch to the event store.                                               |
+| [rollbackIncrementalImport](#rollbackincrementalimport) | Drops an incremental import batch.                                                                    |
+
+### init
+
+Initializes a database.
```
Comment (albertov05) [inline docs/api-reference.md:439]:
```suggestion
Initializes the database.
```

File: docs/api-reference.md [hunk 7]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+import createEventStoreAdapter from 'resolve-eventstore-xxx'
+
+const eventStoreAdapter = createEventStoreAdapter(options)
+
+await eventStoreAdapter.init()
+```
+
+### drop
+
+Drops a database.
```
Comment (albertov05) [inline docs/api-reference.md:453]:
```suggestion
Drops the database.
```

File: docs/api-reference.md [hunk 8]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+
+#### Example
+
+```js
+await eventStoreAdapter.drop()
+```
+
+### dispose
+
+Disconnects from a database and disposes unmanaged resources.
```
Comment (albertov05) [inline docs/api-reference.md:463]:
```suggestion
Disconnects from the database and disposes unmanaged resources.
```

File: docs/api-reference.md [hunk 9]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+  timestamp: Date.now(),
+  payload: {
+    name: 'user-name'
+  }
+})
+```
+
+### loadEvents
+
+Gets an array of events and the next cursor from the store based on the specified filtering criteria.
```
Comment (albertov05) [inline docs/api-reference.md:495]:
```suggestion
Gets an array of events and the next cursor from the store based on the specified filter criteria.
```

File: docs/api-reference.md [hunk 10]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+
+#### Example
+
+```js
+await eventStoreAdapter.unfreeze()
+```
+
+### isFrozen
+
+Gets a boolean value that indicates whether the database is frozen.
```
Comment (albertov05) [inline docs/api-reference.md:605]:
```suggestion
Gets a Boolean value that indicates whether the database is frozen.
```

File: docs/api-reference.md [hunk 11]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
+### saveSnapshot
+
+Creates or updates a snapshot.
+
+##### Arguments
+
+| Argument Name | Description                             |
+| ------------- | --------------------------------------- |
+| snapshotKey   | A unique key in the table of snapshots. |
+| content       | A snapshot in the text format.          |
```
Comment (albertov05) [inline docs/api-reference.md:653]:
```suggestion
| content       | A snapshot in text format.          |
```

File: docs/api-reference.md [hunk 12]
```diff
@@ -405,18 +405,432 @@ await sideEffects.scheduleCommand(
...
 | ------------------------------------- | ----------------------------------------------------------------------------- |
 | [build](#build)                       | Builds an application.                                                        |
-| [start](#start)                       | Runs an application.                                                     |
+| [start](#start)                       | Runs an application.                                                          |
 | [watch](#watch)                       | Runs an application in **watch** mode. (Watch application files for changes.) |
 | [runTestcafe](#runtestcafe)           | Runs TestCafe tests.                                                          |
-| [merge](#merge)                       | Merges modules and application configurations into a single object.                  |
-| [stop](#stop)                         | Stops an application.                                                 |
+| [merge](#merge)                       | Merges modules and application configurations into a single object.           |
```
Comment (albertov05) [inline docs/api-reference.md:832]:
```suggestion
| [merge](#merge)                       | Merges module and application configurations into a single object.           |
```

File: packages/adapters/eventstore-adapters/README.md [hunk 13]
```diff
@@ -36,86 +12,7 @@ Available adapters:
   Used to store events in a PostgreSQL.
 - [resolve-eventstore-postgresql-serverless](./resolve-eventstore-postgresql-serverless)  
    Used to store events in a PostgreSQL Serverless.
```
Comment (albertov05) [inline packages/adapters/eventstore-adapters/README.md:14]: MySQL, PostgreSQL and PostgreSQL Serverless are all database names, right? If so, is it technically correct if we say:

Line 34/10:
`Used to store events in a MySQL database.`

Line 36/12:
`Used to store events in a PostgreSQL database.`

Line 38/14:
`Used to store events in a PostgreSQL Serverless database.`

Comment (EugeniyBurmistrov) [reply packages/adapters/eventstore-adapters/README.md:14]: fixed

File: docs/api-reference.md (hunk 3 above)
Comment (EugeniyBurmistrov) [reply docs/api-reference.md:410]: It is "reSolve". I fixed all occurences of wrong capitalization.

**Review by EugeniyBurmistrov — APPROVED** [review]

File: docs/api-reference.md (hunk 3 above)
Comment (albertov05) [reply docs/api-reference.md:410]: Thank you.

**Pull request merged**